app/controllers/products_controller.rb

```diff
@@ -31,7 +31,7 @@ def create
     end
   end
   
-
+  
   def edit
```

Comment:
i added the "render_404" functionality into the private method `def find_product`, so the "show" and "edit" methods should just become stubs -- just in case you see the two versions in the merge!